### PR TITLE
New logger trunk

### PR DESCRIPTION
**Pull request by MBoldyrev**
### Description of the Change

This trunk contains the new logger implementation fully integrated into the project. Additionally, the configuration file parsing was rewritten and related libraries (fmt & spdlog) were updated.

### Benefits

New logger features, hidden libs & better config parser.

### CODE REVIEW

**Comment by MBoldyrev** [comment]
Commits 2f09018, 4241a1e, e940c49 and d1f2be94e were accidentally merged from #2072 without being squashed.

**Comment by neewy** [comment]
Can you please also write some docs about verbosity level here?..

**Comment by MBoldyrev** [comment]
> Looks good but please fix `mst_fuzz` and `status_fuzzing` and `torii_fuzzing` targets.

These targets are outdated due to missing CI checks for them, and it has nothing to do with the new logger. So I tried to fix them in a separate PR: please see #2136.

File: irohad/validation/impl/stateful_validator_impl.cpp [hunk 1]
```diff
@@ -54,7 +55,7 @@ namespace iroha {
         const shared_model::interface::types::TransactionsCollectionType &txs,
         ametsuchi::TemporaryWsv &temporary_wsv,
         validation::TransactionsErrors &transactions_errors_log,
-        const logger::Logger &log,
+        const logger::LoggerPtr & /*log*/,
```
Comment (igor-egorov) [inline irohad/validation/impl/stateful_validator_impl.cpp]: mb remove that parameter?

File: test/framework/integration_framework/integration_test_framework.hpp [hunk 2]
```diff
@@ -86,6 +90,7 @@ namespace integration_framework {
      * @param block_store_path - specifies path where blocks will be stored
      * @param proposal_waiting - timeout for next proposal appearing
      * @param block_waiting - timeout for next committed block appearing
+     * @param log_manager - log manager
```
Comment (igor-egorov) [inline test/framework/integration_framework/integration_test_framework.hpp]: 👍

File: test/module/irohad/torii/torii_queries_test.cpp [hunk 3]
```diff
@@ -54,7 +55,8 @@ using ErrorQueryType =
 class ToriiQueriesTest : public testing::Test {
  public:
   virtual void SetUp() {
-    runner = std::make_unique<ServerRunner>(ip + ":0");
+    runner = std::make_unique<ServerRunner>(ip + ":0",
+                                            getTestLogger("ServerRunner"  ));
```
Comment (igor-egorov) [inline test/module/irohad/torii/torii_queries_test.cpp]:
```suggestion
                                            getTestLogger("ServerRunner"));
```

File: test/module/libs/crypto/keys_manager_test.cpp [hunk 4]
```diff
@@ -48,7 +49,8 @@ class KeyManager : public ::testing::Test {
       "00576e02f23c8c694c322796cb3ef494829fdf484f4b42312fb7d776fbd5123b"s;
   const std::string prikey =
       "36f028580bb02cc8272a9a020f4200e346e276ae664e45ee80745574e2f5ab80"s;
-  KeysManagerImpl manager = KeysManagerImpl(filepath);
+  const logger::LoggerPtr keysManagerLogger = getTestLogger("KeysManager");
```
Comment (igor-egorov) [inline test/module/libs/crypto/keys_manager_test.cpp]:
```suggestion
  const logger::LoggerPtr kKeysManagerLogger = getTestLogger("KeysManager");
```
or
```suggestion
  const logger::LoggerPtr keys_manager_logger = getTestLogger("KeysManager");
```

File: irohad/main/application.hpp [hunk 5]
```diff
@@ -299,6 +300,10 @@ class Irohad {
 
   std::unique_ptr<ServerRunner> torii_server;
   std::unique_ptr<ServerRunner> internal_server;
+
+  logger::LoggerManagerTreePtr log_manager_;  ///< application root log manager
```
[the remark this answers is not in the document]
Comment (MBoldyrev) [reply irohad/main/application.hpp]: It is the doxygen syntax for succeeding comments: http://www.doxygen.nl/manual/docblocks.html#memberdoc.

File: iroha-cli/interactive/impl/interactive_transaction_cli.cpp [hunk 6]
```diff
@@ -169,13 +170,19 @@ namespace iroha_cli {
         const std::string &creator_account,
         const std::string &default_peer_ip,
         int default_port,
-        const std::shared_ptr<iroha::model::ModelCryptoProvider> &provider)
+        const std::shared_ptr<iroha::model::ModelCryptoProvider> &provider,
+        logger::LoggerManagerTreePtr responce_handler_log_manager,
```
[the remark this answers is not in the document]
Comment (MBoldyrev) [reply iroha-cli/interactive/impl/interactive_transaction_cli.cpp]: thenx

File: irohad/validation/impl/stateful_validator_impl.cpp (hunk 1 above)
Comment (MBoldyrev) [reply irohad/validation/impl/stateful_validator_impl.cpp]: okay

File: test/framework/integration_framework/integration_test_framework.hpp (hunk 2 above)
Comment (MBoldyrev) [reply test/framework/integration_framework/integration_test_framework.hpp]: self-explaining names are great!

**Pull request merged**